code/datums/components/supermatter_crystal.dm

```diff
@@ -4,6 +4,8 @@
 	var/datum/callback/tool_act_callback
 	///Callback used by the SM to get the damage and matter power increase/decrease
 	var/datum/callback/consume_callback
+	// Typecache for similarly handled objects in attackby_proc
+	var/list/attackby_hit_typecache
```

Comment:
```suggestion
	// A whitelist of items that can interact with the SM without dusting the user
	var/static/list/sm_item_whitelist = typecacheof(list(
		/obj/item/melee/roastingstick,
		/obj/item/toy/crayon/spraycan,
	))
```

Try this instead.  You want your list to be properly named and for it to be static. (meaning it won't ever change)

Reply:
yeah that makes sense, I should of thought of that